### PR TITLE
[Model] [Quantization] Support deepseek_v3 w8a8 fp8 block-wise quantization

### PR DESCRIPTION
**Pull request by mgoin**
Adapted from https://github.com/sgl-project/sglang/pull/2575. Thank you @HandH1998

### CODE REVIEW

File: vllm/model_executor/layers/linear.py
Comment (robertgshaw2-neuralmagic) [inline vllm/model_executor/layers/linear.py:630]: Do you know why this is needed?

Also, do you know why it is only needed for `MergedColumnParallel` and not `QKVParallel`

**Review by robertgshaw2-redhat — APPROVED** [review]
Approving to unblock for model release. As a follow up, we should take a look at removing the changes to `linear.py`, so that we can keep a separation of concerns between the parameter and layer

**Comment by simon-mo** [comment]
Local testing result
```
 pytest -v -s test_block_fp8.py -k 'test_w8a8_block_fp8_fused_moe'
===================================================== test session starts ======================================================
platform linux -- Python 3.11.11, pytest-8.3.4, pluggy-1.5.0 -- /home/eecs/xmo/miniconda3/envs/vllm-brewster/bin/python
cachedir: .pytest_cache
rootdir: /data/xmo/vllm
configfile: pyproject.toml
plugins: anyio-4.7.0
collected 203 items / 198 deselected / 5 selected                                                                              

test_block_fp8.py::test_w8a8_block_fp8_fused_moe[1-4608-7168-256-1-block_size0-dtype0-0] WARNING 12-26 15:18:38 fused_moe.py:374] Using default MoE config. Performance might be sub-optimal! Config file not found at /data/xmo/vllm/vllm/model_executor/layers/fused_moe/configs/E=256,N=4608,device_name=NVIDIA_H200,dtype=fp8_w8a8.json
out.sum()=tensor(152., device='cuda:0', dtype=torch.bfloat16)
ref_out.sum()=tensor(35.2500, device='cuda:0', dtype=torch.bfloat16)
PASSED
test_block_fp8.py::test_w8a8_block_fp8_fused_moe[7-4608-7168-256-1-block_size1-dtype1-0] out.sum()=tensor(-403456., device='cuda:0', dtype=torch.bfloat16)
ref_out.sum()=tensor(-403456., device='cuda:0', dtype=torch.bfloat16)
PASSED
test_block_fp8.py::test_w8a8_block_fp8_fused_moe[83-4608-7168-256-1-block_size2-dtype2-0] out.sum()=tensor(-5408., device='cuda:0', dtype=torch.bfloat16)
ref_out.sum()=tensor(-3648., device='cuda:0', dtype=torch.bfloat16)
PASSED
test_block_fp8.py::test_w8a8_block_fp8_fused_moe[512-4608-7168-256-1-block_size3-dtype3-0] out.sum()=tensor(473088., device='cuda:0', dtype=torch.bfloat16)
ref_out.sum()=tensor(485376., device='cuda:0', dtype=torch.bfloat16)
PASSED
test_block_fp8.py::test_w8a8_block_fp8_fused_moe[2048-4608-7168-256-1-block_size4-dtype4-0] out.sum()=tensor(-4259840., device='cuda:0', dtype=torch.bfloat16)
ref_out.sum()=tensor(-4227072., device='cuda:0', dtype=torch.bfloat16)
PASSED

================================= 5 passed, 198 deselected in 300.20s (0:05:00) =================================
```

**Review by robertgshaw2-redhat — APPROVED** [review]

**Pull request merged**
